src/api/auth.ts

```diff
@@ -135,221 +148,206 @@ function authenticatedUser(name: string, groups: string[] = []): any {
+      let orgIndex = auth.organizations.findIndex(org => org.name === organization);
+      let teamIndex = auth.organizations[orgIndex].teams.findIndex(t => t.name === team);
+      let userIndex = auth.organizations[orgIndex].teams[teamIndex]
+        .members.findIndex(u => u === username);
+      auth.organizations[orgIndex].teams[teamIndex].members.splice(userIndex, 1);
+      resolve(auth);
     });
 }
 
 export function deleteUserFromOrganization(
-  req: express.Request, res: express.Response): express.Response | void {
-    let { username, organization } = req.body;
-    let auth = getAuth();
-    let orgIndex = auth.organizations.findIndex(org => org.name === organization);
-    let usrIndex = auth.organizations[orgIndex].members.findIndex(u => u.name === username);
-    auth.organizations[orgIndex].members.splice(usrIndex, 1);
-    writeJsonFile(getAuthPath(), auth).then(
-      () => res.status(200).json({ success: 'User successfully deleted from an organization.' }));
+  username: string, organization: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+      let auth = getAuth();
+      let orgIndex = auth.organizations.findIndex(org => org.name === organization);
+      let usrIndex = auth.organizations[orgIndex].members.findIndex(u => u.name === username);
+      auth.organizations[orgIndex].members.splice(usrIndex, 1);
+      resolve(auth);
+    });
 }
 
-export function deleteOrganization(
-  req: express.Request, res: express.Response): express.Response | void {
-    let organization = req.body.organization;
+export function deleteOrganization(organization: string): Promise<string> {
+  return new Promise((resolve, reject) => {
     let auth = getAuth();
     let index = auth.organizations.findIndex(org => org.name === organization);
     auth.organizations.splice(index, 1);
-    writeJsonFile(getAuthPath(), auth).then(() => {
-      res.status(200).json({ success: 'Organization successfully deleted.' });
-    });
+    resolve(auth);
+  });
 }
 
 export function changeUserRole(
-  req: express.Request, res: express.Response): express.Response | void {
-    let { username, organization, role } = req.body;
-    let auth = getAuth();
-    let orgIndex = auth.organizations.findIndex(org => org.name === organization);
-    let usrIndex = auth.organizations[orgIndex].members.findIndex(u => u.name === username);
-    auth.organizations[orgIndex].members[usrIndex].role = role;
-    writeJsonFile(getAuthPath(), auth).then(() => {
-      res.status(200).json({ success: 'User role successfully updated.' });
+  username: string, organization: string, role: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+      let auth = getAuth();
+      let orgIndex = auth.organizations.findIndex(org => org.name === organization);
+      let usrIndex = auth.organizations[orgIndex].members.findIndex(u => u.name === username);
+      auth.organizations[orgIndex].members[usrIndex].role = role;
```

Comment:
It shouldn't happened. A case like that has to be prevented from the api. But one extra check won't hurt just in case. I'll add index verification...